server/src/server.js

```diff
@@ -675,10 +696,23 @@ exports.simpleResponse = simpleResponse;
 exports.errorResponse = errorResponse;
 
 linker.init().then(() => {
-  app.listen(config.port);
-  console.info(`server listening on http://localhost:${config.port}/`);
-  contentApp.listen(config.contentPort);
-  console.info(`content server listening on http://localhost:${config.contentPort}/`);
+  if (config.useVirtualHosts) {
+    const mainapp = express();
+    const siteName = config.siteOrigin.split(":")[0];
+    const contentName = config.contentOrigin.split(":")[0];
+    mainapp.use(vhost(siteName, app));
+    mainapp.use(vhost(contentName, contentApp));
+    mainapp.listen(config.port);
+    mainapp.get("/", function (req, res) {
+      res.send("ok");
+    });
+    console.info(`virtual host server listening on http://localhost:${config.port}`);
+  } else {
+    app.listen(config.port);
+    console.info(`server listening on http://localhost:${config.port}/`);
+    contentApp.listen(config.contentPort);
+    console.info(`content server listening on http://localhost:${config.contentPort}/`);
+  }
```

Comment:
I'm betting we could do `mainapp.use(vhost("localhost:10080"))` and `mainapp.use(vhost("localhost:10081"))` and it would work, and then we wouldn't need two sets of logic for two kinds of dispatching.  Then I also assume we can call `mainapp.listen()` twice with the two ports.